File: src/Domain/FastShop.BackOffice.Domain/Entities/Order.cs
using FastShop.BackOffice.Domain.Enum;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace FastShop.BackOffice.Domain.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime CreateAt { get; set; }
        public DateTime UpdateAt { get; set; }
        public PaymentTypeEnum PaymentType { get; set; }
        public OrderStatusEnum Status { get; set; }
        public int ClientId { get; set; }

        [ForeignKey("ClientId")]
        public Client Client { get; set; }
        public List<OrderItem> Items { get; set; }
        public decimal Amount => Items.Sum(x => x.TotalPartial());

    }
}

File: src/Presentation/FastShop.BackOffice.Web/ClientApp/src/app/types/Order.ts
import { Client } from './client';
import { OrderItem } from './OrderItem';

export class Order {
  id: number;
  createAt: Date;
  updateAt: Date;
  paymentType: number;
  status: number;
  clientId: number;
  client: Client
  items: OrderItem[];
  amount: number;
}
